fw/http_msg.c

```diff
@@ -1553,13 +1551,15 @@ __tfw_http_msg_expand_from_pool(TfwHttpResp *resp, const TfwStr *str,
 
 			n_copy = min(n_copy, skb_room);
 
-			r = tfw_http_msg_alloc_from_pool(mit, pool, n_copy);
-			if (unlikely(r))
+			mit->curr_ptr = tfw_http_msg_alloc_from_pool(it, pool,
+								     n_copy);
+			if (unlikely(!mit->curr_ptr))
 				return r;
```

Comment:
We can't return 'r' here, since it was not changed during previous `tfw_http_msg_alloc_from_pool` function call. We should return -ENOMEM here, since we don't save error code from 'tfw_http_msg_alloc_from_pool' now. Or we can add "int *error" to `tfw_http_msg_alloc_from_pool` arguments and pass "r" there.